### PR TITLE
Convert AuthenticationCoordinatedViewController and AuthenticationErrorFeedbackAction

### PR DESCRIPTION
**Pull request by billypchan**
Convert AuthenticationCoordinatedViewController and AuthenticationErrorFeedbackAction to Swift.
Change the protocol method to non-optional to handle errors in each view controller.

### discussion
- [ ] we should implement default `displayError` ? All of them are empty now.

### CODE REVIEW

File: Wire-iOS.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -2349,10 +2349,10 @@
...
+		A955D3D223F54AC400304851 /* AuthenticationCoordinatedViewController.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = AuthenticationCoordinatedViewController.swift; sourceTree = "<group>"; };
 		A961319F23E81CC300194608 /* ParticipantDeviceHeaderView.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = ParticipantDeviceHeaderView.swift; sourceTree = "<group>"; };
 		A9614EF723C38C810012A718 /* ConnectRequestsViewController.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = ConnectRequestsViewController.swift; sourceTree = "<group>"; };
 		A96316A423F2B10500AFDB71 /* ProfileTitleViewSnapshotTests.swift */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.swift; path = ProfileTitleViewSnapshotTests.swift; sourceTree = "<group>"; };
-		A963643123E2E52F00190992 /* AudioTrackPlayer+Private.h */ = {isa = PBXFileReference; lastKnownFileType = sourcecode.c.h; path = "AudioTrackPlayer+Private.h"; sourceTree = "<group>"; };
```
Comment (johnxnguyen) [inline Wire-iOS.xcodeproj/project.pbxproj]: Is this intentional?

File: Wire-iOS/Sources/Authentication/Coordinator/AuthenticationCoordinator.swift [hunk 2]
```diff
@@ -146,7 +146,7 @@ extension AuthenticationCoordinator: AuthenticationStateControllerDelegate {
             return
         }
 
-        guard let stepViewController = interfaceBuilder.makeViewController(for: newState) else {
+        guard var stepViewController = interfaceBuilder.makeViewController(for: newState) else {
```
Comment (johnxnguyen) [inline Wire-iOS/Sources/Authentication/Coordinator/AuthenticationCoordinator.swift]: Does this need to be var?

File: Wire-iOS.xcodeproj/project.pbxproj (hunk 1 above)
Comment (billypchan) [reply Wire-iOS.xcodeproj/project.pbxproj]: XCode fixes this for the missing file.

File: Wire-iOS/Sources/Authentication/Coordinator/AuthenticationCoordinator.swift (hunk 2 above)
Comment (billypchan) [reply Wire-iOS/Sources/Authentication/Coordinator/AuthenticationCoordinator.swift]: objc version allow `stepViewController.authenticationCoordinator = self` later but Swift does not.

**Pull request merged**